nifi-commons/nifi-security-utils/src/test/java/org/apache/nifi/security/util/crypto/BcryptSecureHasherTest.java

```diff
@@ -0,0 +1,334 @@
+
+        BcryptSecureHasher bcryptSH = new BcryptSecureHasher();
+
+        // Act
+        String hashHex = bcryptSH.hashHex(input);
+
+        // Assert
+        assertEquals(EXPECTED_HASH_HEX, hashHex);
+    }
+
+    @Test
+    void testShouldFormatBase64() {
+        // Arrange
+        String input = "This is a sensitive value";
+
+        final String EXPECTED_HASH_BASE64 = "JDJhJDEyJFJrakVZUS9SYkVEeVlVS2VTMERxYi5YUmlhNTRNTjVsWkRRUkNXWHRRbExpbUdmaWNad2hx";
+
+        BcryptSecureHasher bcryptSH = new BcryptSecureHasher();
+
+        // Act
+        String hashB64 = bcryptSH.hashBase64(input);
+
+        // Assert
+        assertEquals(EXPECTED_HASH_BASE64, hashB64);
+    }
+
+    @Test
+    void testShouldHandleNullInput() {
+        // Arrange
+        List<String> inputs = Arrays.asList(null, "");
+
+        final String EXPECTED_HASH_HEX = "";
+        final String EXPECTED_HASH_BASE64 = "";
+
+        BcryptSecureHasher bcryptSH = new BcryptSecureHasher();
+
+        final List<String> hexResults = new ArrayList<>();
+        final List<String> b64Results = new ArrayList<>();
+
+        // Act
+        for (final String input : inputs) {
+            String hashHex = bcryptSH.hashHex(input);
+            hexResults.add(hashHex);
+
+            String hashB64 = bcryptSH.hashBase64(input);
+            b64Results.add(hashB64);
+        }
+
+        // Assert
+        hexResults.forEach(result -> assertEquals(EXPECTED_HASH_HEX, result));
+        b64Results.forEach(result -> assertEquals(EXPECTED_HASH_BASE64, result));
+    }
+
+    /**
+     * This test can have the minimum time threshold updated to determine if the performance
+     * is still sufficient compared to the existing threat model.
+     */
+    @EnabledIfSystemProperty(named = "nifi.test.performance", matches = "true")
+    @Test
+    void testDefaultCostParamsShouldBeSufficient() {
+        // Arrange
+        int testIterations = 100;
+        byte[] inputBytes = "This is a sensitive value".getBytes();
+
+        BcryptSecureHasher bcryptSH = new BcryptSecureHasher();
+
+        final List<String> results = new ArrayList<>();
+        final List<Long> resultDurations = new ArrayList<>();
+
+        // Act
+        for (int i = 0; i < testIterations; i++) {
+            long startNanos = System.nanoTime();
+            byte[] hash = bcryptSH.hashRaw(inputBytes);
+            long endNanos = System.nanoTime();
+            long durationNanos = endNanos - startNanos;
+
+            String hashHex = new String(Hex.encode(hash));
+
+            results.add(hashHex);
+            resultDurations.add(durationNanos);
+        }
+
+        // Assert
+        final long MIN_DURATION_NANOS = 75_000_000; // 75 ms
+        assertTrue(Collections.min(resultDurations) > MIN_DURATION_NANOS);
+        assertTrue(resultDurations.stream().mapToLong(Long::longValue).sum() / testIterations > MIN_DURATION_NANOS);
+    }
```

Comment:
As some algorithms are no longer used for framework encryption operations, this test method can be removed.

Reply:
Removed it